src/lib/voiceSearchHelper/types.ts

```diff
@@ -0,0 +1,34 @@
+export type Status =
+  | 'initial'
+  | 'askingPermission'
+  | 'waiting'
+  | 'recognizing'
+  | 'finished'
+  | 'error';
+
+export type VoiceListeningState = {
```

Comment:
To make sure I understand: are this state properties necessary for all Voice solutions, or only for our default implementation? In other words, should we accept unknown properties in the state?

Reply:
We could accept arbitrary props here as well, but these are the props we are using in our own component, documentation & template, so at least those are required (and should always be able to be present, at least with a static value).

I don't really see how to combine arbitrary props with a set of required ones, but will try out

Reply:
I don't know if there is any standard spec or something, but `VoiceListeningState` and `Status` just came from our default implementation which is based on SpeechRecognition web api.
These fit at least our implementation and this new custom helper one. We could see later if we need to accept unknown properties for other third party helpers.

Reply:
Then I would suggest making it a generic that extends the required properties: `VoiceListeningState<TState>`.

Reply:
Recapping our discussions earlier, we decided that we don't need arbitrary states at the moment and this shouldn't block folding this into the other flavors.